cx-portal/src/components/shared/frame/Header/index.tsx

```diff
@@ -43,28 +45,59 @@ export const Header = ({ main, user }: { main: Tree[]; user: string[] }) => {
   const menu = addTitle(main) || []
 
   return (
-    <header>
-      <MainNavigation items={menu} component={NavLink}>
-        <Logo />
-        <div className="d-flex">
-          <Button
-            size="small"
-            color="secondary"
-            variant="contained"
-            onClick={() => {
-              window.open(
-                `${document.location.origin}/documentation/`,
-                'documentation',
-                'noreferrer'
-              )
-            }}
-            sx={{ backgroundColor: 'white', marginRight: '16px' }}
-          >
-            {t('pages.help')}
-          </Button>
-          <UserInfo pages={user} />
-        </div>
-      </MainNavigation>
-    </header>
+    <>
+      <header>
+        <MainNavigation items={menu} component={NavLink}>
+          <Logo />
+          <div className="d-flex">
+            <div
+              onClick={() => {
+                let ele = document.getElementById('new-search')
+                if (ele && ele.style.visibility === 'hidden') {
+                  ele.style.visibility = 'visible'
+                } else if (ele) {
+                  ele.style.visibility = 'hidden'
+                }
+              }}
```

Comment:
what's the reason we don't use a react useState here? I would expect something like:

`const [searchVisible, setSearchVisible] = useState<boolean>(false)`
...
`onClick={() => { setSearchVisible(!searchVisible) }`
...
`searchVisible && <SomeItem...>`

or use Redux, if they're not in the same component.
If there is no other solution at least rewrite to:

`ele.style.visibility = ele && ele.style.visibility === 'hidden' ? 'visible' : 'hidden'`

Reply:
I had written like this - ele.style.visibility = ele && ele.style.visibility === 'hidden' ? 'visible' : 'hidden'

But got sonar cloud issue where it had suggested to re write as I have done now.

Reply:
I guess React state management won't work here. The element is not present in the current component but it is available in the DOM. So using JS, changing css property from `hidden` to `visible`.

Reply:
If the element is in another component then it's better to use Redux similar like we do with the overlays. I suggest to create a file `src/features/control/appear.ts` with an enum `Appear.SEARCH` and actions `show(item)`, `hide(item)`, `toggle(item)` and `appear(item, visible: boolean)` to control the state of this and similar future components and then invoke:

`onClick={() => dispatch(toggle(Appear.SEARCH))}`

and in the other components react to state changes using a selector

`const visible = useSelector(appearSearchSelector)`

`visible && <SomeItem...>`

Note that this might seem over engineered at first glance. However the problem is again that we start to introduce a dirty pattern that will be copied and multiplied in the future. Let's do it the clean way from the beginning.